src/subtype.c

```diff
@@ -1665,6 +1665,8 @@ JL_DLLEXPORT int jl_obvious_subtype(jl_value_t *x, jl_value_t *y, int *subtype)
                 assert(vy != JL_VARARG_NONE && istuple && iscov);
                 jl_value_t *a1 = (vx != JL_VARARG_NONE && i >= npx - 1) ? vxt : jl_tparam(x, i);
                 jl_value_t *b = jl_unwrap_vararg(jl_tparam(y, i));
+                if (jl_is_typevar(b) && var_occurs_inside(y, (jl_tvar_t*)b, 0, 1))
```

Comment:
Isn't this equivalent to:
```suggestion
                 if (jl_is_typevar(b))
```
Which seems not what you meant to write (though valid).

Reply:
Or it's maybe actually:
```suggestion
                if (jl_is_typevar(b) && var_occurs_inside(jl_tparam(y, i), (jl_tvar_t*)b, 0, 1))
```
(e.g. it's also checking that the Vararg is not wrapped in a UnionAll with `b` as the variable—which seems reasonable to do, but should never appear in practice due to normalization, so again, equivalent to the above)